main.py

```diff
@@ -1,53 +1,66 @@
 import logging
 import os
 import re
+import json
+import dpath.util
 
 import boto3
 import requests
 from botocore.exceptions import ClientError
 
 logging.basicConfig(
     level=logging.DEBUG,
-    format='%(asctime)s %(message)s',
+    format="%(asctime)s %(message)s",
     handlers=[logging.StreamHandler()]
 )
 
-COVERAGE_FILE = os.getenv("INPUT_COVERAGE_FILE", "coverage.txt")
-S3_PATH = os.getenv("INPUT_S3_PATH")
-BADGE_NAME = os.getenv("INPUT_BADGE_NAME")
-UPLOAD_COVERAGE_FILE = os.getenv("INPUT_UPLOAD_COVERAGE_FILE", False)
-BUCKET_NAME = os.getenv("INPUT_BUCKET_NAME")
-ACCESS_KEY = os.getenv("INPUT_AWS_ACCESS_KEY")
-SECRET_KEY = os.getenv("INPUT_AWS_SECRET_KEY")
-COVERAGE_PERCENTAGE_REGEX = os.getenv("INPUT_COVERAGE_PERCENTAGE_REGEX")
+COVERAGE_FILE = os.getenv("INPUT_COVERAGE_FILE", "")
+BADGE_NAME = os.getenv("INPUT_BADGE_NAME", "")
+UPLOAD_COVERAGE_FILE = os.getenv("INPUT_UPLOAD_COVERAGE_FILE", "False").lower() in ("true", "1", "t", "y", "yes")
+BUCKET_NAME = os.getenv("INPUT_BUCKET_NAME", "")
+ACCESS_KEY = os.getenv("INPUT_AWS_ACCESS_KEY", "")
+SECRET_KEY = os.getenv("INPUT_AWS_SECRET_KEY", "")
+COVERAGE_PERCENTAGE_REGEX_OR_PATH = os.getenv("INPUT_COVERAGE_PERCENTAGE_REGEX", "")
 
+BRIGHT_GREEN_PERCENTAGE_THRESHOLD = 90
 GREEN_PERCENTAGE_THRESHOLD = 80
+YELLOW_GREEN_PERCENTAGE_THRESHOLD = 70
+YELLOW_PERCENTAGE_THRESHOLD = 60
 ORANGE_PERCENTAGE_THRESHOLD = 50
 
 
-def compute_coverage():
+def get_coverage() -> int:
     """
-    Compute coverage percentage by parsing coverage file with a regex.
+    Get coverage percentage from parsed coverage file.
 
-    :return: The computed coverage
+    :return: The extracted coverage
     :rtype: int
     """
     logging.info(
-        f"Computing coverage from {COVERAGE_FILE} using "
-        f"REGEX {COVERAGE_PERCENTAGE_REGEX}")
-    with open(COVERAGE_FILE) as f:
-        return int(
-            float(
-                re.findall(COVERAGE_PERCENTAGE_REGEX, f.read())[0].replace(
-                    "%",
-                    "")
+        f"Extracting coverage from {COVERAGE_FILE} on "
+        f"regex/path {COVERAGE_PERCENTAGE_REGEX_OR_PATH}"
+    )
+    if COVERAGE_FILE.endswith("json"):
```

Comment:
Let's make the type configurable in order to avoid the if else condition. It will be known a priori based on the env var being set.

Reply:
@anmarkoulis can you elaborate on this?